### DISCUSSION TITLE
How should I write a nested function template like {{urldecode({{base64_decode('A')}})}}?

### DISCUSSION THREAD

**Discussion by Y4er** [projectdiscovery/nuclei#1622] [Q&A] [answered]
### Please describe your feature request:

How should I write a nested function template
### Describe the use case of this feature:

like {{urldecode({{base64_decode('A')}})}}

**Comment by forgedhallpass** [comment, marked as answer]
Hello @Y4er,

You don't need DSL expression separators, within an existing DSL context, hence the correct way is:
`{{url_decode(base64_decode('your input here'))}}`

Also, please note, the correct function name is `url_decode`.
See https://nuclei.projectdiscovery.io/templating-guide/helper-functions/

**Reply by Y4er** [reply]
Does it work in extractor too? I need to calculate cookies based on http response.

**Reply by ehsandeep** [reply]
For extractor, you can do something like this `{{url_decode(base64_decode('{{test}}'))}}` where `test` is dynamic variable name defined for extractor.

**Reply by hktalent** [reply]
Current DSL support functions：@Y4er
cat nuclei/v2/pkg/operators/common/dsl/dsl.go|grep "\": makeDsl"|sed 's/              "//g'|sed 's/".*//g'
```
len
to_upper
to_lower
repeat
replace
replace_regex
trim
trim_left
trim_right
trim_space
trim_prefix
trim_suffix
reverse
base64
gzip
gzip_decode
zlib
zlib_decode
date
time
timetostring
base64_py
base64_decode
url_encode
url_decode
hex_encode
hex_decode
html_escape
html_unescape
md5
sha256
sha1
mmh3
contains
concat
regex
remove_bad_chars
rand_char
rand_base
rand_text_alphanumeric
rand_text_alpha
rand_text_numeric
rand_int
rand_ip
generate_java_gadget
unix_time
wait_for
compare_versions
print_debug
to_number
to_string
```

**Comment by hktalent** [comment]
Hello @Y4er,你的这个需求，我可以帮你实现哦
